store/postgres/migrations/2019-07-20-195916_remove_subgraph/up.sql

```diff
@@ -0,0 +1,122 @@
+-- List all metadata entities for a set of subgraph deployments.
+-- The entities are returned as a table of (id, entity) pairs, in no particular
+-- order. The list contains all the entities that are reachable through
+-- associations from a SubgraphDeployment, and include the SubgraphDeployment
+-- itself.
+create or replace function
+  list_deployment_entities(ids varchar[])
+returns table(id varchar, entity varchar) as
+$$
+begin
+  create temp table tmp_ids
+      on commit drop as
+  select * from unnest(ids) as id;
+  create index tmp_ids_id on tmp_ids(id);
+
+  -- Put the ids of all dynamic data sources for the
+  -- subgraph deployments in ids into tmp_dds
+  create temp table tmp_dds
+      on commit drop as
+  select dds.id
+    from subgraphs.entities dds
+   where dds.entity = 'DynamicEthereumContractDataSource'
+     and dds.data->'deployment'->>'data' = any(ids);
+   create index tmp_dds_id on tmp_dds(id);
+
+   -- The metadata entities for a subgraph deployment consist of two
+   -- distinct groups of entities:
+   --   * static metadata, whose id starts with the id of the
+   --     subgraph deployment. Subgraph deployment ids are 46 chars long
+   --   * metadata for dynamic data sources, whose ids start with
+   --     the id of the dynamic data source. Dynamic data source ids are
+   --     40 chars long
+   return query
+   select e.id, e.entity
+     from subgraphs.entities e, tmp_ids
+    where substring(e.id, 1, 46) = tmp_ids.id
+   union all
+   select e.id, e.entity
+     from subgraphs.entities e, tmp_dds
+    where substring(e.id, 1, 40) = tmp_dds.id;
```

Comment:
Clever to use the id prefix on all entities.

Reply:
I started with carefully chasing pointers through the metadata, but that turned out to be incredibly slow. This works much faster.